tests/workflows/basic_tool_calling_node/tests/test_workflow.py

```diff
@@ -0,0 +1,256 @@
+            text='{\n    "arguments": {\n        "location": "Miami",\n        "unit": "metric"\n    },\n    "id": "call_7115tNTmEACTsQRGwKpJipJK",\n    "name": "get_current_weather",\n    "state": "FULFILLED"\n}',  # noqa: E501
+            role="assistant",
+            content=None,
+            source=None,
+        ),
+        ChatMessage(
+            text="Result from get_current_weather: The current temperature is 22°C (71.6°F).",
+            role="tool",
+            content=None,
+            source=None,
+        ),
+        ChatMessage(
+            text="The current temperature in San Francisco is 22°C (71.6°F).",
+            role="assistant",
+            content=None,
+            source=None,
+        ),
+    ]
+
+    first_call = vellum_adhoc_prompt_client.adhoc_execute_prompt_stream.call_args_list[0]
+    assert first_call.kwargs == {
+        "ml_model": "gpt-4o-mini",
+        "input_values": [
+            PromptRequestStringInput(key="question", type="STRING", value="What's the weather like in San Francisco?")
+        ],
+        "input_variables": [
+            VellumVariable(
+                id=str(first_call_input_id),
+                key="question",
+                type="STRING",
+                required=None,
+                default=None,
+                extensions=None,
+            )
+        ],
+        "parameters": DEFAULT_PROMPT_PARAMETERS,
+        "blocks": [
+            ChatMessagePromptBlock(
+                block_type="CHAT_MESSAGE",
+                state=None,
+                cache_config=None,
+                chat_role="SYSTEM",
+                chat_source=None,
+                chat_message_unterminated=None,
+                blocks=[
+                    RichTextPromptBlock(
+                        block_type="RICH_TEXT",
+                        state=None,
+                        cache_config=None,
+                        blocks=[
+                            PlainTextPromptBlock(
+                                block_type="PLAIN_TEXT", state=None, cache_config=None, text="You are a weather expert"
+                            )
+                        ],
+                    )
+                ],
+            ),
+            ChatMessagePromptBlock(
+                block_type="CHAT_MESSAGE",
+                state=None,
+                cache_config=None,
+                chat_role="USER",
+                chat_source=None,
+                chat_message_unterminated=None,
+                blocks=[
+                    RichTextPromptBlock(
+                        block_type="RICH_TEXT",
+                        state=None,
+                        cache_config=None,
+                        blocks=[
+                            VariablePromptBlock(
+                                block_type="VARIABLE", state=None, cache_config=None, input_variable="question"
+                            )
+                        ],
+                    )
+                ],
+            ),
```

Comment:
[nit] Going with the chat history approach I mentioned above, I would expect a variable block here that references the chat_history variable